src/main/java/uk/ac/ebi/eva/contigalias/datasource/ENAAssemblyDataSource.java

```diff
@@ -57,23 +65,35 @@ public ENAAssemblyDataSource(ENABrowserFactory factory,
     public Optional<AssemblyEntity> getAssemblyByAccession(String accession) throws IOException {
         ENABrowser enaBrowser = factory.build();
         enaBrowser.connect();
-
-        AssemblyEntity assemblyEntity;
-        try (InputStream stream = enaBrowser.getAssemblyReportInputStream(accession)) {
-            if (Objects.isNull(stream)) {
-                logger.warn("Could not fetch assembly report from ENA for accession : " + accession);
+        try {
+            String dirPath = enaBrowser.getAssemblyDirPath(accession);
+            FTPFile ftpFile = enaBrowser.getAssemblyReportFile(dirPath, accession);
+            String ftpFilePath = dirPath + ftpFile.getName();
+            Path downloadFilePath = Paths.get(asmFileDownloadDir, ftpFile.getName());
```

Comment:
Optional: There is a discussion to be had as to whether we want this downloaded file should be permanent and if so, should we defer to the standard path where we currently download these assembly reports (via other automation)?

Reply:
**Optional:** Suggest refactoring the download code to a separate function

Reply:
Would the webservice have access to NFS mounts ? 
I would not use the one we currently use because they have been sometime modified.

Reply:
Actually I'm wondering if the files should be deleted once we're done processing it so that we're certain not to run our of space in the `/tmp` directory.

Reply:
added the code to delete file after being read